shared/macros-oval.jinja

```diff
@@ -0,0 +1,88 @@
+{{%- set none_exist = "_none_exist" -%}}
+{{%- endif %}}
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    <metadata>
+      <title>{{{ msg or rule_title }}}</title>
+      <affected family="unix">
+        <platform>multi_platform_all</platform>
+      </affected>
+      <description>{{{ msg or rule_title }}}</description>
+    </metadata>
+    <criteria comment="{{{ application }}} is configured correctly"
+    operator="OR">
+        {{{ oval_line_in_file_criteria(path, parameter) }}}
+        {{%- if none_exist %}}
+        {{{ oval_line_in_file_criteria(path, parameter, none_exist) }}}
+        {{%- endif %}}
+    </criteria>
+  </definition>
+
+  {{{ oval_line_in_file_test(path, parameter) }}}
+  {{{ oval_line_in_file_object(path, prefix_regex, parameter, separator_regex, value) }}}
+  {{{ oval_line_in_file_state(value, exact_match) }}}
+  {{%- if none_exist %}}
+  {{{ oval_line_in_file_test(path, parameter, none_exist) }}}
+  {{{ oval_line_in_file_object(path, prefix_regex, parameter, separator_regex, value, none_exist) }}}
+  {{%- endif %}}
+
+</def-group>
+{{%- endmacro %}}
+
+{{%- macro oval_line_in_file_criteria(path='', parameter='', none_exist='') -%}}
+{{%- if none_exist %}}
+<criterion comment="Check absence of {{{ parameter }}} in {{{ path }}}"
+{{%- else %}}
+<criterion comment="Check {{{ parameter }}} in {{{ path }}}"
+{{%- endif %}}
+  test_ref="test_{{{ rule_id }}}{{{ none_exist }}}" />
+{{%- endmacro %}}
+
+{{%- macro oval_line_in_file_test(path='', parameter='', none_exist='') -%}}
+  {{%- if none_exist %}}
+  <ind:textfilecontent54_test check="all" check_existence="none_exist"
+  comment="tests the absence of {{{ parameter }}} setting in the {{{ path }}} file"
+  {{%- else %}}
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  comment="tests the value of {{{ parameter }}} setting in the {{{ path }}} file"
+  {{%- endif %}}
+  id="test_{{{ rule_id }}}{{{ none_exist }}}" version="1">
+  <ind:object object_ref="obj_{{{ rule_id }}}{{{ none_exist }}}" />
+  {{%- if not none_exist %}}
+  <ind:state state_ref="state_{{{ rule_id }}}" />
+  {{%- endif %}}
+  </ind:textfilecontent54_test>
+{{%- endmacro %}}
+
+{{%- macro oval_line_in_file_object(path='', prefix_regex='^[\s]*', parameter='', separator_regex='[\s]+', value='', none_exist='') -%}}
+  <ind:textfilecontent54_object id="obj_{{{ rule_id }}}{{{ none_exist }}}" version="1">
+    <ind:filepath>{{{ path }}}</ind:filepath>
+    {{%- if none_exist %}}
+    <ind:pattern operation="pattern match">{{{ prefix_regex }}}{{{ parameter }}}{{{ separator_regex }}}</ind:pattern>
+    {{%- else %}}
+    <ind:pattern operation="pattern match">{{{ prefix_regex }}}{{{ parameter }}}{{{ separator_regex }}}(.*)[\s]*(?:|(?:#.*))?$</ind:pattern>
+    {{%- endif %}}
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+{{%- endmacro %}}
+
+{{%- macro oval_line_in_file_state(value='', exact_match='') -%}}
```

Comment:
It would be better to not complicate it and have this macro only with one parameter, e.g. `line_regex`.

Reply:
In this case I would like to avoid having to construct the regex before, maybe the `exact_match` keyword was not the best option here. I'll make a new commit soon to change the approach and we can continue the discussion.

Reply:
I think we still need the second parameters.

I see where you are going, the intention is to check whether the subexpression in the matched object should be the only value, or the parameter can have other multiple values.
And `value` can also be a regex, so the goal is to remove the burden from content writer to also have to provide `value` with regex accounting for other extra things.

Having an argument `exact_match` can be confusing. Why would I want to mach exactly with a regex?
I suggest to rename it to `multi_value`, or something similar.

Note: The `<subexpression>`'s operation is `pattern match`, so we are doing pattern matching all the time, :stuck_out_tongue: .

Reply:
I changed to multi_value. I didn't understand exactly what the `<subexpression>` means here, maybe I have to dig up in the OVAL standard a bit.